### DISCUSSION TITLE
Adding external video source inside a conference room

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5409] [FAQs]
Can I add an external video source e.g., a YouTube video inside a conference room?

**Comment by USAMAWIZARD** [comment]
Yes its something possible , If you have a direct mp4 URL of the video you can First add it as a stream src.



![](https://attachment.freshdesk.com/inline/attachment?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6NDcyMzMyMjA1NDcsImRvbWFpbiI6ImFudG1lZGlhLmZyZXNoZGVzay5jb20iLCJhY2NvdW50X2lkIjoxMjY0ODM4fQ.tRXj81of8GZYzwJshV4ZMHrOOHozIoRC2KuBnV8RSe0)

  

If you don't have a direct MP4 file URL and You want to use YouTube for steaming src you can either download and then upload the video to AMS or follow the below procedure for alternate solution.

  

you will need to start an ffmpeg command that will fetch stream from youtbe and stream it ant media using SRT

  

Streaming YouTube video without uploading it to Ant Media Server

```
python3 -m pip install --force-reinstall https://github.com/yt-dlp/yt-dlp/archive/master.tar.gz

ffmpeg -re -stream_loop -10 -i  "$(yt-dlp  -g -f best https://www.youtube.com/watch?v=xvFZjo5PgG0)" -c:v libx264 -x264-params keyint=80:scenecut=0  -preset fast -tune zerolatency -f mpegts  srt://AMS_IP:4200?streamid=WebRTCAppEE/stream11
```


  

Once you have a stream up and running on Ant Media you can call  [this API](https://antmedia.io/rest/#/BroadcastRestService/addStreamToTheRoom)  to add it to a Room with particular RoomId

  

curl -X PUT  [https://IP_ADDRESS;5443/LiveApp/rest/v2/broadcasts/conference-rooms/room1/add?streamId=stream1](https://ip_address%3B5443/LiveApp/rest/v2/broadcasts/conference-rooms/room1/add?streamId=stream1)
